httpbakery/agent/agent.go

```diff
@@ -0,0 +1,150 @@
+/*
+Package agent implements the agent login mechanism. The agent login
+mechanism provides a form of interaction that can be used in
+situations where interaction with the user is not possible.
+
+An agent login works as follows:
```

Comment:
Most of this comment is actually not necessary if you just want to use this package.
I wonder if most of this comment (useful as it is) should perhaps be an internal
comment, and this package doc should just outline how to use the package.

e.g.

	Package agent enables non-interactive (agent) login using macaroons.
	To enable agent authorization with a given httpbakery.Client c against
	a given third party discharge server URL u:
	
	    agent.SetCookie(c, u, agentUsername, &c.Key.Public)
	    c.VisitWebPage = agent.VisitWebPage(c)

writing this down like this makes me think that perhaps the API
is more complex than it needs to be.

How about:

    // SetUpAuth enables non-interactive authorization on the given client.
    // Whenever the client is directed to obtain a discharge macaroon
    // from a 3rd party under the given URL, the given username
    // will be provided along with the client's public key.
    // SetUpAuth does this by setting a cookie in c.Client.Jar and
    // setting c.VisitWebPage to a direct web page fetcher.
    func SetUpAuth(c *httpbakery.Client, u *url.URL, username string)

 ?